src/components/editor/PropertyTemplateOutline.jsx

```diff
@@ -180,8 +180,13 @@ export class PropertyTemplateOutline extends Component {
   handleClick = (property) => (event) => {
     event.preventDefault()
     let newOutput = this.state.output
-
-    this.fulfillRTPromises(this.resourceTemplatePromises(property.valueConstraint.valueTemplateRefs))
+    const templateRefList = []
+    if(hasValueTemplateRef(property)) {
+      property.valueConstraint.valueTemplateRefs.map((row) => {
```

Comment:
I'm a bit confused about what this is doing. Is the intent of this PR to add a check before sending a value to `this.resourceTemplatePromises()`?

Is this code essentially creating a new variable to hold an exact copy of an array? If so, I'm not sure `map` is the clearest solution unless there is a function you want to apply to every member of the array. The intent would be clearer, in my judgment, if you simplified this to:

```javascript
const templateRefList = hasValueTemplateRef(property) ? property.valueConstraint.valueTemplateRefs : []
```

Would that work?

Reply:
Or, if you add the safe navigation/"Elvis"/optional chaining feature via babel, you can simplify this further to:

```javascript
const templateRefList = property?.valueConstraint?.valueTemplateRefs || []
```

Reply:
Thanks! You're right that using `map` over-complicates the `templateRefList` variable setting; I need to read-up on the babel chaining feature you mention but I went ahead and pushed up your first suggestion.